### PR TITLE
Offnode scanning

### PR DESCRIPTION
**Pull request by bryanro92**
### Which issue this PR addresses:

https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/9749860/

### What this PR does / why we need it:

Receives the `AzureSecPackVSATenantId` value from [RP-Config](https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/RP-Config/pullrequest/4740810?_a=files), and lays down a config file for azsecpack to enable off node scanning. This is a requirement of gov cloud/fedramp 

### Test plan for issue:

Deployed to INT and confirmed off node scanning is now reporting as expected https://portal.microsoftgeneva.com/s/267A40A2 

### Is there any documentation that needs to be updated for this PR?

docs in wiki  https://msazure.visualstudio.com/AzureRedHatOpenShift/_wiki/wikis/ARO.wiki/169104/AzSecPack-Off-Node-Scanning

### CODE REVIEW

File: pkg/deploy/config.go
Comment (mjudeikis) [inline pkg/deploy/config.go:44]: What `VSA` stands for?

**Review by mjudeikis — CHANGES_REQUESTED** [review]
The fact we dont have parameters files and templates regenerated this looks like it is missing something.
Needs more work for full review

**Comment by troy0820** [comment]
@bryanro92 Could you link the RP-Config PR that has this configuration?  Also, when LGTM, can you also squash to 1 commit as the unit of work is small and doesn't need to be broken into multiple commits.

File: pkg/deploy/config.go
Comment (bryanro92) [reply pkg/deploy/config.go:44]: Vulnerability Scanning and Analytics/triage (VSA) it is internal Microsoft team that owns the scanning we are enrolling into.

**Comment by bryanro92** [comment]
> The fact we dont have parameters files and templates regenerated this looks like it is missing something.
> Needs more work for full review

Working on `make generate` files, hitting an error in go environment working on resolving

File: pkg/deploy/generator/resources_rp.go
Comment (m1kola) [inline pkg/deploy/generator/resources_rp.go:896]: Is it ready for review or should this PR be in the draft state?

File: pkg/deploy/config.go
Comment (troy0820) [inline pkg/deploy/config.go:44]: Your test is failing because the parameter you have is `azSecPackVSATenantID` instead of what's in the config here as `azureSecPackVSATenantId`.  Even though it's in the config, it can not resolve the parameter set on the config.

Comment (troy0820) [reply pkg/deploy/config.go:44]: For consistency, it may make sense to use `azureSecPackVSATenantId` in the `deploy/rp-production-parameters.json`

File: deploy/rp-production-parameters.json
Comment (troy0820) [inline deploy/rp-production-parameters.json:25]: Comment on the config file not matching parameter causing golang unit test to fail.

File: deploy/rp-production.json
Comment (troy0820) [inline deploy/rp-production.json:419]: Same as above

File: pkg/deploy/generator/resources_rp.go
Comment (troy0820) [inline pkg/deploy/generator/resources_rp.go:380]: Same comment as above

File: pkg/deploy/generator/templates_rp.go
Comment (troy0820) [inline pkg/deploy/generator/templates_rp.go:39]: Same as above.

**Review by troy0820 — CHANGES_REQUESTED** [review]
Once these changes are in, this should push you forward.  I'm not sure if a deploy to INT is necessary to test the changes (@mjudeikis, @m1kola, @jewzaam).

File: pkg/deploy/config.go
Comment (bryanro92) [reply pkg/deploy/config.go:44]: changing to azure, updated rp-config too https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/RP-Config/commit/bcf3556c3a9793e25cd3c937791873a6b5d01c98?refName=refs%2Fheads%2Fazsecpack_offnode

File: deploy/rp-production-parameters.json
Comment (troy0820) [inline deploy/rp-production-parameters.json:5]: This should be lowercase to be consistent with the other params and to match the json from the "config".

File: pkg/deploy/generator/resources_rp.go
Comment (troy0820) [inline pkg/deploy/generator/resources_rp.go:380]: Same as above

File: pkg/deploy/generator/templates_rp.go
Comment (troy0820) [inline pkg/deploy/generator/templates_rp.go:39]: lowercase as listed in above comment.

**Review by troy0820 — CHANGES_REQUESTED** [review]
These changes should allow the golang test to pass as well as pass e2e.

File: pkg/deploy/generator/resources_rp.go
Comment (m1kola) [inline pkg/deploy/generator/resources_rp.go:894]: I'm not sure that I understand: Where do we install the agents itself

Comment (bryanro92) [reply pkg/deploy/generator/resources_rp.go:894]: The binary that runs is the `azsec` binary that is already part of vmss template. I am just adding additional configuration. I've followed instructions (below) that is supposed to auto-"magically" download the Qualys plugin needed for azsec process to perform scan/
image

Comment (bryanro92) [reply pkg/deploy/generator/resources_rp.go:894]: I am following the recommended method, but I am not exactly sure specific details of how Qualys plugin does get downloaded. I assume that the Managed Identity that is allowlisted reports being online via `azsec` process to central scanning server, and then that process downloads Qualys plugin and is able to perform scans.  

https://microsoft.sharepoint.com/teams/VulnerableNodes/SitePages/Acquiring-the-private-Qualys-plugin.aspx (require msft VM or SAW)

**Review by troy0820 — COMMENTED** [review]
If we need to deploy to INT to test this change and see the azureSecPackVSA on the vmss, maybe link the results of that deployment.  
Also, link the RP-Config PR that links that param included in the configuration.

**Comment by bryanro92** [comment]
Will deploy PR to INT and verify working as expected thru log queries. INT build job is queued https://msazure.visualstudio.com/AzureRedHatOpenShift/_build/results?buildId=44453844&view=logs&j=627439e9-18e7-568f-4f44-799be08ba1a7

**Comment by bryanro92** [comment]
deployed to INT https://msazure.visualstudio.com/AzureRedHatOpenShift/_build/results?buildId=44459471&view=logs&j=ac5ecb5a-030b-5f61-b782-cded586d007c 

confirmed this enables off-node scanning on INT logs here: https://portal.microsoftgeneva.com/s/267A40A2

**Review by mjudeikis — DISMISSED** [review]

**Comment by mjudeikis** [comment]
> deployed to INT https://msazure.visualstudio.com/AzureRedHatOpenShift/_build/results?buildId=44459471&view=logs&j=ac5ecb5a-030b-5f61-b782-cded586d007c
> 
> confirmed this enables off-node scanning on INT logs here: https://portal.microsoftgeneva.com/s/267A40A2

Let me know when its ok to merge after your tests

**Comment by bryanro92** [comment]
> Let me know when its ok to merge after your tests

Working as expected, ready for merge.

**Comment by mjudeikis** [comment]
@bryanro92 please rebase

**Pull request closed without merging**
